src/test/url-handling.test.js

```diff
@@ -103,49 +109,81 @@ describe('selectedThread', function() {
-    const { getState } = _getStoreWithURL({ search: '' }, profileWithThreads);
-    expect(urlStateReducers.getThreadOrder(getState())).toEqual([0, 2, 1]);
-    expect(urlStateReducers.getHiddenThreads(getState())).toEqual([]);
-  });
+  describe('global tracks', function() {
+    it('creates tracks without any set search parameters', function() {
+      const { getState } = initWithSearchParams('');
+      expect(getHumanReadableTracks(getState())).toEqual([
+        'show [thread GeckoMain process] SELECTED',
+        'show [thread GeckoMain tab]',
+        '  - show [thread DOM Worker]',
+        '  - show [thread Style]',
+      ]);
+    });
 
-  it('can reorder the threads ', function() {
-    const { getState } = _getStoreWithURL(
-      { search: '?threadOrder=1-2-0' },
-      profileWithThreads
-    );
-    expect(urlStateReducers.getThreadOrder(getState())).toEqual([1, 2, 0]);
-    expect(urlStateReducers.getHiddenThreads(getState())).toEqual([]);
-  });
+    it('can reorder global tracks ', function() {
+      const { getState } = initWithSearchParams('?globalTrackOrder=1-0');
+      expect(getHumanReadableTracks(getState())).toEqual([
+        'show [thread GeckoMain tab]',
+        '  - show [thread DOM Worker]',
+        '  - show [thread Style]',
+        'show [thread GeckoMain process] SELECTED',
+      ]);
+    });
 
-  it('can hide the threads', function() {
-    const { getState } = _getStoreWithURL(
-      { search: '?hiddenThreads=1-2' },
-      profileWithThreads
-    );
-    expect(urlStateReducers.getThreadOrder(getState())).toEqual([0, 2, 1]);
-    expect(urlStateReducers.getHiddenThreads(getState())).toEqual([1, 2]);
-  });
+    it('can hide tracks', function() {
+      const { getState } = initWithSearchParams('?hiddenGlobalTracks=1');
+      expect(getHumanReadableTracks(getState())).toEqual([
+        'show [thread GeckoMain process] SELECTED',
+        'hide [thread GeckoMain tab]',
+        '  - show [thread DOM Worker]',
+        '  - show [thread Style]',
+      ]);
+    });
 
-  it('will not accept invalid threads in the thread order', function() {
-    const { getState } = _getStoreWithURL(
-      { search: '?threadOrder=0-8-2-a-1' },
-      profileWithThreads
-    );
-    expect(urlStateReducers.getThreadOrder(getState())).toEqual([0, 2, 1]);
-    expect(urlStateReducers.getHiddenThreads(getState())).toEqual([]);
+    it('will not accept invalid tracks in the thread order', function() {
+      const { getState } = initWithSearchParams('?globalTrackOrder=1-0');
+      expect(urlStateReducers.getGlobalTrackOrder(getState())).toEqual([1, 0]);
+    });
+
+    it('will not accept invalid hidden threads', function() {
+      const { getState } = initWithSearchParams(
+        '?hiddenGlobalTracks=0-8-2-a&thread=1'
+      );
+      expect(urlStateReducers.getHiddenGlobalTracks(getState())).toEqual(
+        new Set([0])
+      );
```

Comment:
I remember issues with comparing Sets with Jest, that's why I did https://github.com/devtools-html/perf.html/blob/7a8d8d0dd57a284e17526944b14dd444cbd00256/src/test/fixtures/custom-assertions.js#L7-L14

I don't remember what the issue was though :(

Reply:
I tried out some assertions locally and they all worked as I expected.